az-digital/az_quickstart
# Fixes #2313 Update Citation Style Configuration

## Description
The `seboettg/citeproc-php` package has updated in its `2.6.0` release to begin using the `citation-style-language/locales` and `citation-style-language/styles` packages for its CSL rules. These packages were not previously available on packagist and it used to be a manual step to install them.

Previously, Quickstart was using `academicpuma/locales` (an older copy of the same locale files that was available on packagist) for its locale information, to prevent needing to install them manually.

Now that these styles are available on packagist, we may want to consider refactoring how we include the styles. This PR updates the Publication content type to use `citation-style-language/locales` and modifies our configuration style entities to prefer referencing CSL styles in the `citation-style-language/styles` package by name rather than including the xml directly.

Modifications are made to the style configuration form to allow either referencing a CSL style by name, or by custom XML. Most of our managed styles now refer to the `citation-style-language/styles` by reference. Bluebook Law Review currently still uses custom XML as the `citation-style-language/styles` version of bluebook law review does not support bibliographies The custom version of quickstart's Bluebook Law Review managed style has been modified to support bibliographies.

_Spike: Feedback is desired on the best way to include references to the stylesheets, given there are a very large number of options in `citation-style-language/styles`; dropdown built via cached introspection of the files, specifying name manually, etc._

## Related issues
#2313 

## How to test

- enable `az_publication`
- create some publications
- verify citations are still rendered correctly
- visit `/admin/config/az-quickstart/settings/az-publication/az_citation_style`
- create a new style or edit one
- For the new style, enter a stylesheet name or custom CSL (see [Citation Style Language Repository](https://github.com/citation-style-language/styles) for examples. Enter stylesheet names without the `.csl` extension, e.g. `modern-language-association`
- change the default site style at `/admin/config/az-quickstart/settings/az-publication` to your style and verify they work correctly

## Types of changes


### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

### Drupal core
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

### Drupal contrib projects
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

## Checklist
